src/rust/src/decoder/mod.rs

```diff
@@ -0,0 +1,97 @@
+//! CEA 708 decoder
+//!
+//! This module provides a CEA 708 decoder as defined by ANSI/CTA-708-E R-2018
+
+use crate::bindings::*;
+
+const CCX_DTVCC_MAX_PACKET_LENGTH: u8 = 128;
+
+/// Process cc data to generate dtvcc packet
+///
+/// data has following format:
+///
+/// | Name | Size |
+/// | --- | --- |
+/// | cc_valid | 1 byte |
+/// | cc_type | 1 byte |
+/// | data | 2 bytes |
+///
+/// # Safety
+///
+/// dtvcc and data must not be null pointers
+#[no_mangle]
+pub unsafe extern "C" fn dtvcc_process_cc_data(
+    dtvcc: *mut dtvcc_ctx,
+    data: *const ::std::os::raw::c_uchar,
+) {
+    let dtvcc = Dtvcc::new(dtvcc);
+    let mut ctx = &mut *dtvcc.ctx;
+    let cc_valid = *data;
+    let cc_type = *data.add(1);
+
+    match cc_type {
+        // type 0 and 1 are for CEA 608 data and are handled before calling this function
+        // valid types for CEA 708 data are only 2 and 3
+        2 => {
+            println!("[CEA-708] dtvcc_process_data: DTVCC Channel Packet Data");
```

Comment:
Are these debug leftovers? I haven't tried it but doesn't this produce thousands of lines on console?

Reply:
These lines are for logs. I have temporarily added them as println. I have to figure out logging

Reply:
I know, but this makes the program unusable. This can't be merged as is. 
It's not that rust doesn't have good logging crates :-)

Reply:
Yes, I'll fix this